### PR TITLE
Adds a Node interface to Sale

### PR DESCRIPTION
**Pull request by ashfurrow**
Continues work from #794. Paired with @orta. This is required to get Relay working with Emission for sales.

### CODE REVIEW

**Comment by ashfurrow** [comment]
I will address the test failures on Monday.

**Comment by ashfurrow** [comment]
This will probably be useful for getting the tests to pass: https://medium.com/@boriscoder/the-hidden-power-of-jest-matchers-f3d86d8101b0

File: schema/sale/index.js [hunk 1]
```diff
@@ -12,6 +13,8 @@ import { connectionFromArraySlice, connectionDefinitions } from "graphql-relay"
 import { amount } from "schema/fields/money"
 import { exclude } from "lib/helpers"
 import { map } from "lodash"
```
Comment (mzikherman) [inline schema/sale/index.js]: You can add `has` here and avoid the import below.

File: schema/object_identification.js [hunk 2]
```diff
@@ -47,6 +47,7 @@ const SupportedTypes = {
     "./partner",
     "./partner_show",
     "./show",
+    "./sale",
```
Comment (mzikherman) [inline schema/object_identification.js]: 👍

File: schema/sale/index.js [hunk 3]
```diff
@@ -59,6 +62,8 @@ const saleArtworkConnection = connectionDefinitions({
 
 const SaleType = new GraphQLObjectType({
   name: "Sale",
+  interfaces: [NodeInterface],
+  isTypeOf: obj => _.has(obj, "is_auction") && _.has(obj, "currency"),
```
Comment (mzikherman) [inline schema/sale/index.js]: 👍 seems reasonable!

**Comment by mzikherman** [comment]
Nice!

I've experienced that before- adding the `NodeInterface` and `isTypeOf` sometimes requires updating mocked sale fixtures with those newly required fields.

**Comment by ashfurrow** [comment]
Thanks! I've written up some notes and will add what I learned to the wiki for next time.

**Pull request merged**